test/api.js

```diff
@@ -76,6 +76,15 @@ describe('api', function() {
       });
     });
 
+    it('should NOT compile empty data string', function(done) {
+      sass.render({
+        data: ''
+      }, function(error) {
+        assert.equal(error.message, 'Data context created with empty source string');
```

Comment:
I am kind of ambivalent about this:

- we are pretty thin wrapper here and I wouldn't really introduce own `node-sass` error messages if `libsass` already handles the situation
- I am not sure that having two kinds of contexts (data and file) are good design. They are actually the same structures differing just by one extra field may be very error prone. A data context with `source_string` set to zero is indistinguishable from the file context - we have to use two sets of functions to handle the distinction. We also have ambiguous situation when both `file` and `data` are set - there is a special meaning for this situation as well (see #914).

We need some work on the libsass side to have this ironed out in my opinion.